#! /bin/sh /usr/share/dpatch/dpatch-run
## ubuntu_fname_quote_percent.dpatch by  <martin.pitt@ubuntu.com>
##
## DP: Escape '%' in file names.
## DP: http://bugzilla.gnome.org/show_bug.cgi?id=433829,
## DP: https://launchpad.net/bugs/258083

@DPATCH@
Index: f-spot.git/src/Utils/UriUtils.cs
===================================================================
--- f-spot.git.orig/src/Utils/UriUtils.cs	2009-10-29 14:18:29.150757347 +0000
+++ f-spot.git/src/Utils/UriUtils.cs	2009-10-29 14:19:08.265757117 +0000
@@ -45,7 +45,7 @@
 			return new Uri (builder.ToString ());
 		}
 
-		static char[] CharsToQuote = { ';', '?', ':', '@', '&', '=', '$', ',', '#' };
+		static char[] CharsToQuote = { ';', '?', ':', '@', '&', '=', '$', ',', '#', '%' };
 		// NOTE: this was copied from mono's System.Uri where it is internal.
 		public static string EscapeString (string str, bool escapeReserved, bool escapeHex, bool escapeBrackets) 
 		{
